src/components/motor_remap/MotorRemapCanvas.js

```diff
@@ -0,0 +1,272 @@
+            if (this._spinMotorCallback) {
+                this._spinMotorCallback(this._motorIndexToSpinOnMouseDown);
+            }
+        }
+    }
+
+    _markMotors()
+    {
+        var ctx = this._ctx;
+        var droneConfiguration = this._droneConfiguration;
+        var config = this._config;
+        var motors = config[droneConfiguration].Motors;
+
+        if (-1 == this._motorIndexToSpinOnMouseDown) {
+            for (let i = 0; i < this.readyMotors.length; i++) {
+                var motorIndex = this.readyMotors[i];
+                ctx.beginPath();
+                ctx.arc(motors[motorIndex].x, motors[motorIndex].y, config[droneConfiguration].PropRadius, 0, 2 * Math.PI);
+                ctx.closePath();
+                ctx.fillStyle = config.MotorReadyColor;
+                ctx.fill();
+            }
+
+            var mouseHoverMotorIndex = this._getMouseHoverMotorIndex();
+            if (mouseHoverMotorIndex != -1 && !this.readyMotors.includes(mouseHoverMotorIndex)) {
+                ctx.beginPath();
+                ctx.arc(motors[mouseHoverMotorIndex].x, motors[mouseHoverMotorIndex].y, config[droneConfiguration].PropRadius, 0, 2 * Math.PI);
+                ctx.closePath();
+                ctx.fillStyle = config.MotorMouseHoverColor;
+                ctx.fill();
+            }
+        } else {
+            var mouseHoverMotorIndex = this._getMouseHoverMotorIndex();
+            var spinningMotor = this._motorIndexToSpinOnMouseDown;
+
+            var motors = config[droneConfiguration].Motors;
+
+            for (let i = 0; i < motors.length; i++) {
+                ctx.fillStyle = config.MotorReadyColor;
+                if (i == spinningMotor) {
+                    ctx.fillStyle = config.MotorSpinningColor;
+                } else if (i == mouseHoverMotorIndex) {
+                    ctx.fillStyle = config.MotorMouseHoverColor;
+                }
+
+                ctx.beginPath();
+                ctx.arc(motors[i].x, motors[i].y, config[droneConfiguration].PropRadius, 0, 2 * Math.PI);
+                ctx.closePath();
+                ctx.fill();
+            }
+        }
+    }
+
+    _getMouseHoverMotorIndex()
+    {
+        var x = this._mouse.x;
+        var y = this._mouse.y;
+
+        var result = -1;
+        var currentDist = Number.MAX_SAFE_INTEGER;
+        var droneConfiguration = this._droneConfiguration;
+        var motors = this._config[droneConfiguration].Motors;
+
+        for (let i = 0; i < motors.length; i++) {
+            var dist = Math.sqrt((x - motors[i].x) * (x - motors[i].x) + (y - motors[i].y) * (y - motors[i].y));
+            if (dist < this._config[droneConfiguration].PropRadius && dist < currentDist) {
+                currentDist = dist;
+                result = i;
+            }
+        }
+
+        return result;
+    }
+
+    _drawMotors()
+    {
+        var ctx = this._ctx;
+        var droneConfiguration = this._droneConfiguration;
+        var config = this._config;
```

Comment:
I've seen you use a lot of auxiliar variables. I suppose is question of how each one codes, but when the auxiliar variable does nothing (is not a transformation of another) I prefer to use directly the `this._xxxxxx` in the code. Is an easy way to see that the variable is global to the class.